src/overlay005/field_control.c

```diff
@@ -0,0 +1,1092 @@
+            ov8_0224C51C(fieldSystem) == TRUE) {
+
+            sub_0205F56C(fieldSystem->playerAvatar);
+            MapObjectMan_PauseAllMovement(fieldSystem->mapObjMan);
+            return TRUE;
+        }
+    }
+
+    if (input->movement) {
+        sub_0206A9A4(SaveData_Events(fieldSystem->saveData));
+
+        if (Field_ProcessStep(fieldSystem) == TRUE) {
+            return TRUE;
+        }
+    }
+
+    if (input->unused5 == FALSE) {
+        int playerEvent = PLAYER_EVENT_NONE;
+        int direction = sub_02061308(fieldSystem->playerAvatar, input->pressedKeys, input->heldKeys);
+
+        if (inline_0204E650_2(SaveData_Events(fieldSystem->saveData))) {
+            playerEvent |= PLAYER_EVENT_USED_STRENGTH;
+        }
+
+        if (sub_020549A0(Party_GetFromSavedata(fieldSystem->saveData), 127) != 0xff) {
+            playerEvent |= PLAYER_EVENT_USED_WATERFALL;
+        }
+
+        if (sub_02071CB4(fieldSystem, 9) == TRUE && ov9_02250F74(fieldSystem) == TRUE) {
+            playerEvent |= PLAYER_EVENT_DISTORTION_WORLD;
+        }
+
+        if (ov5_021DFDE0(fieldSystem, fieldSystem->playerAvatar, direction, playerEvent) == TRUE) {
+            return TRUE;
+        }
+    }
+
+    if (input->endMovement) {
+        if (Field_CheckWildEncounter(fieldSystem)) {
+            return TRUE;
+        }
+
+        if (Field_CheckSign(fieldSystem) == TRUE) {
+            return TRUE;
+        }
+
+        if (sub_02071CB4(fieldSystem, 9) == TRUE) {
+            int direction = (s8)input->playerDir;
+
+            if (direction == DIR_NONE) {
+                direction = PlayerAvatar_GetDir(fieldSystem->playerAvatar);
+            }
+
+            if (ov9_0224A67C(fieldSystem, direction) == TRUE) {
+                return TRUE;
+            }
+        }
+    }
+
+    if (input->sign && PlayerAvatar_GetDir(fieldSystem->playerAvatar) == input->playerDir) {
+        if (Field_CheckSign(fieldSystem) == TRUE) {
+            return TRUE;
+        }
+    }
+
+    if (input->interact) {
+        int interactCheck;
+        MapObject *object;
+
+        if (sub_02071CB4(fieldSystem, 9) == TRUE) {
+            interactCheck = Field_DistortionInteract(fieldSystem, &object);
+        } else {
+            interactCheck = sub_0203CA40(fieldSystem, &object);
+        }
+
+        if (interactCheck == TRUE) {
+            if (sub_0205F588(fieldSystem->playerAvatar) == TRUE) {
+                sub_0205F5E4(fieldSystem->playerAvatar, PlayerAvatar_GetDir(fieldSystem->playerAvatar));
+            }
+
+            if (sub_02062950(object) != 0x9) {
+                sub_0203E880(fieldSystem, sub_02062960(object), object);
+            } else {
+                sub_0203E880(fieldSystem, 0, object);
+            }
+
+            return TRUE;
+        }
+
+        int distortionCheck = sub_0205F108(fieldSystem->playerAvatar);
+
+        if (distortionCheck == AVATAR_DISTORTION_STATE_NONE || distortionCheck == AVATAR_DISTORTION_STATE_ACTIVE) {
```

Comment:
Is this branch always taken?

Reply:
Bump

Reply:
Nope, there's a handful of actual possible outcomes of `sub_0205F180()`.

Reply:
Update the definition of `enum AvatarDistortionState` then to include the other possible values